crates/biome_css_parser/src/token_source.rs

```diff
@@ -1,117 +1,242 @@
-use crate::lexer::{Lexer, Token};
+use crate::lexer::{CssLexContext, CssLexer, CssReLexContext};
 use crate::CssParserOptions;
-use biome_css_syntax::CssSyntaxKind::{EOF, TOMBSTONE};
+use biome_css_syntax::CssSyntaxKind::EOF;
 use biome_css_syntax::{CssSyntaxKind, TextRange};
 use biome_parser::diagnostic::ParseDiagnostic;
-use biome_parser::prelude::TokenSource;
+use biome_parser::lexer::{BufferedLexer, LexContext};
+use biome_parser::prelude::{BumpWithContext, NthToken, TokenSource};
 use biome_parser::token_source::Trivia;
 use biome_rowan::TriviaPieceKind;
-
-pub(crate) struct CssTokenSource<'source> {
-    lexer: Lexer<'source>,
-    trivia: Vec<Trivia>,
-    current: CssSyntaxKind,
-    current_range: TextRange,
-    preceding_line_break: bool,
+use std::collections::VecDeque;
+
+pub(crate) struct CssTokenSource<'src> {
+    lexer: BufferedLexer<'src, CssLexer<'src>>,
+
+    /// List of the skipped trivia. Needed to construct the CST and compute the non-trivia token offsets.
+    pub(super) trivia_list: Vec<Trivia>,
+    /// Cache for the non-trivia token lookahead. For example for the source `let a = 10;` if the
+    /// [TokenSource]'s currently positioned at the start of the file (`let`). The `nth(2)` non-trivia token,
+    /// as returned by the [TokenSource], is the `=` token but retrieving it requires skipping over the
+    /// two whitespace trivia tokens (first between `let` and `a`, second between `a` and `=`).
+    /// The [TokenSource] state then is:
+    ///
+    /// * `non_trivia_lookahead`: [IDENT: 'a', EQ]
+    /// * `lookahead_offset`: 4 (the `=` is the 4th token after the `let` keyword)
```

Comment:
I don't think the documentation can be applied here. The documentation contains javascript code. If we can't document it properly for the CSS lexer at this point, we can leave it blank for the time being.